dubbo-registry/dubbo-registry-default/src/test/java/org/apache/dubbo/registry/dubbo/RegistryDirectoryTest.java

```diff
@@ -1014,6 +1014,37 @@ public void test_Notified_acceptProtocol2() {
         List<Invoker<DemoService>> invokers = registryDirectory.list(invocation);
         Assert.assertEquals(2, invokers.size());
     }
+    
+    @Test
+    public void test_Notified_withGroupFilter() {
+    	URL directoryUrl = noMeaningUrl.addParameterAndEncoded(Constants.REFER_KEY, "interface" + service + "&group=group1,group2");
+    	RegistryDirectory directory = this.getRegistryDirectory(directoryUrl);
+    	URL provider1 = URL.valueOf("dubbo://10.134.108.1:20880?methods=getXXX&group=group1&mock=false");

```

Comment:
Do you feel it below mention way could be a better approach 
URL provider1=URL.valueOf("dubbo://10.134.108.1:20880")
               .addParameter("methods", "getXXX")
               .addParameter("group", "group1")
               .addParameter("mock","false");
This might give better readability and might prevent accidental mistake of placing url escape character.